timesketch/frontend-ng/src/components/Explore/AddManualEvent.vue

```diff
@@ -0,0 +1,182 @@
+            v-model="dateFromPicker"
+            mode="dateTime"
+            ref="picker"
+            timezone="UTC"
+            :is-dark="$vuetify.theme.dark"
+            is24hr
+            is-date
+            is-expanded
+          ></date-picker>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col cols="6">
+          <v-text-field :value="message" v-model="message" label="Message" outlined hide-details> </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row>
+        <v-col cols="6">
+          <v-text-field
+            :value="timestampDesc"
+            label="Timestamp Description"
+            v-model="timestampDesc"
+            outlined
+            hide-details
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row v-for="(attribute, index) in attributes" :key="index">
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Name"
+            outlined
+            hide-details
+            v-model="attributes[index].name"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Value"
+            outlined
+            hide-details
+            v-model="attributes[index].value"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn text color="primary" @click="addAttribute" :disabled="isDisabled"> Add Attribute </v-btn>
+        <v-btn text color="primary" @click="clearAndCancel"> Cancel </v-btn>
+        <v-btn text color="primary" @click="submit"> Add Event </v-btn>
+      </v-card-actions>
+    </v-container>
+  </v-card>
+</template>
+
+<script>
+import dayjs from '@/plugins/dayjs'
+import DatePicker from 'v-calendar/lib/components/date-picker.umd'
+import ApiClient from '../../utils/RestApiClient'
+
+export default {
+  props: ['datetimeProp'],
+  components: {
+    DatePicker,
+  },
+  data() {
+    return {
+      message: null,
+      timestampDesc: null,
+      filterTab: null,
+      showPicker: false,
+      attributes: [],
+      isDisabled: false,
+      datetime: this.datetimeProp,
+    }
+  },
+  watch: {
+    datetimeProp: function () {
+      this.datetime = this.datetimeProp
+    },
+  },
+  computed: {
+    dateFromPicker: {
+      set(val) {
+        if (val !== null) this.datetime = dayjs.utc(val).millisecond(0).toISOString()
+      },
+      get() {
+        return this.datetime
+      },
+    },
+  },
+  methods: {
+    setAddAttributeBtn: function (i) {
+      if (this.attributes[i].name === '' || this.attributes[i].value === '') {
+        this.isDisabled = true
+      } else {
+        this.isDisabled = false
+      }
+    },
+    addAttribute: function () {
+      this.attributes.push({ name: '', value: '' })
+      this.isDisabled = true
```

Comment:
You wouldn't need to set isDisabled if you go with the computed property. (same for L154)

Reply:
```
<v-btn text color="primary" @click="attributes.push({ name: '', value: '' })" :disabled="isDisabled"> 
    Add Attribute 
</v-btn>
```